feat: 대시보드 하단부 구현 완료

siddltkfkd opened the pull request:
1. 실시간 전력량(w) top 10을 가로형 막대그래프로 출력
2. 이번달, 저번달, 오늘 총 전력 사용량(kwh) 출력

Commit:
feat: 대시보드 하단부 구현 완료

haedanui commented on src/main/java/live/ioteatime/frontservice/service/ElectricityService.java:
서비스를 따로 만드신 이유가 있을까요?

siddltkfkd replied:
ElectricityService는 ElectricityAdaptor의 getRealtimeElectricity로 가져온 RealtimeElectricityResponseDto 리스트를 현재 전력량을 기준으로 내림차순으로 정렬한 후, 현재 전력량이 제일 높은 10개의 리스트로 정제하는 기능을 수행합니다.
controller나 adaptor에서 이 기능을 수행하는것 보다 service에서 수행하는것이 올바르다고 판단하여 서비스를 따로 만들었습니다.

haedanui reviewed (approved):
굿

codethestudent commented on src/main/java/live/ioteatime/frontservice/dto/RealtimeElectricityResponseDto.java:
```diff
@@ -0,0 +1,14 @@
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
```
data 어노테이션 보단 사용하는 메서드만 열어 두는게 좋을 거 같습니다 ~ (getter, setter,,)

siddltkfkd replied:
수정하겠습니다

caboooom commented on src/main/resources/templates/index.html:
```diff
@@ -194,6 +245,58 @@ <h4 class="fw-semibold mb-3">목표 요금 재설정</h4>
+    chart.render();
+
+    function fetchDataAndUpdateChart() {
+        fetch('/bottom/top10')
```
url이 `/bottom/top10`이 맞나요?
컨트롤러에는 `/top10`으로 적힌 것 같아서여

siddltkfkd replied:
앗 실수.. 
원래 bottomIndexController를 IndexController에 합치는 과정에서 url을 안고쳤네요

Commit:
refactor: index의 url과 dto의 어노테이션 수정

1. index의 fetch url /bottom/top10을 /top10으로 변경
2. RealtimeElectricityResponseDto의 @Data 어노테이션을 @Getter, @Setter로 변경
